File: 413-arithmetic-slices/arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        dp = [0 for i in range(len(nums))]
        for i in range(2,len(nums)):
            if nums[i-1] * 2 == nums[i-2] + nums[i]:
                dp[i] = dp[i-1] + 1
        print(dp)
        return sum(dp)

File: 413-arithmetic-slices/arithmetic-slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        vector<int>dp(nums.size(),0);
        for(int i=2;i<nums.size();i++){
            if (nums[i]-nums[i-1]==nums[i-1]-nums[i-2]){
                dp[i]=dp[i-1]+1;
            }
        }
        int ans=0;
        for (int i=0;i<dp.size();i++){
            ans+=dp[i];
        }
        return ans;
    }
};